### PR TITLE
[#IR-332]  As a user, I can see My Time for every ticket as a column

### PR DESCRIPTION
**Pull request by Ravi2714**

### CODE REVIEW

File: apps/retrospective/serializers/sprint_task.go [hunk 1]
```diff
@@ -4,30 +4,31 @@ import "time"
...
+	Rating                 int8
+	Estimate               float64
+	TotalPointsEarned      float64
+	PointsEarned           float64
+	Assignee               string // Assignee of the task
+	Owner                  string // Owner of the task across the sprints
+	TaskParticipants       string // Participants of the task across the sprints
+	SprintOwner            string // Owner of the task in the sprint
+	SprintParticipants     string // Participants of the task in the current sprint
+	SprintActiveMemberTime uint   // Time spent on the task by the Current Login Member in the sprint
```
Comment (himanshu-garg-165) [inline apps/retrospective/serializers/sprint_task.go]: Please use a different name here, it looks a bit confusing.

File: apps/retrospective/services/sprint_task_member.go [hunk 2]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
...
+func (service SprintTaskMemberService) AddSprintActiveMember(taskList *retroSerializers.SprintTasksSerializer,
+	memberID uint,
+	retroID string,
+	sprintID string) *retroSerializers.SprintTasksSerializer {
+	db := service.DB
+	sprintMemberTask := retroModels.SprintMemberTask{}
+	sprintMember := retroModels.SprintMember{}
+
+	db.Model(&retroModels.SprintMember{}).
+		Where("sprint_members.deleted_at IS NULL").
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task_member.go]: Please use the `NotDeletedScope` here and create it if not created now.

File: apps/retrospective/services/sprint_task_member.go [hunk 3]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
...
+		Where("sprint_id = ?", sprintID).
+		Where("member_id = ?", memberID).
+		Find(&sprintMember)
+
+	sprintMemberTask.SprintMemberID = sprintMember.ID
+	for _, task := range taskList.Tasks {
+		sprintMemberTask.SprintTaskID = task.ID
+		task.SprintActiveMemberTime = 0
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task_member.go]: No need to do this.

File: apps/retrospective/services/sprint_task_member.go [hunk 4]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
 	return members, http.StatusOK, nil
 }
 
+// AddSprintActiveMember ...
+func (service SprintTaskMemberService) AddSprintActiveMember(taskList *retroSerializers.SprintTasksSerializer,
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task_member.go]: Please use the sprint task service's `List` method and compute the `My Time` like other fields.

File: controllers/v1/sprint_task.go [hunk 5]
```diff
@@ -39,7 +40,7 @@ func (ctrl SprintTaskController) List(c *gin.Context) {
 	}
 
 	tasks, status, err := ctrl.SprintTaskService.List(retroID, sprintID)
-
+	tasks = ctrl.SprintTaskMemberService.AddSprintActiveMember(tasks, userID.(uint), retroID, sprintID)
```
Comment (himanshu-garg-165) [inline controllers/v1/sprint_task.go]: Not the optimized way.

**Review by himanshu-garg-165 — CHANGES_REQUESTED** [review]

File: apps/retrospective/services/sprint_task_member.go [hunk 6]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
...
+	memberID uint,
+	retroID string,
+	sprintID string) *retroSerializers.SprintTasksSerializer {
+	db := service.DB
+	sprintMemberTask := retroModels.SprintMemberTask{}
+	sprintMember := retroModels.SprintMember{}
+
+	db.Model(&retroModels.SprintMember{}).
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]: handle error also.

File: apps/retrospective/services/sprint_task_member.go [hunk 7]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
 	return members, http.StatusOK, nil
 }
 
+// AddSprintActiveMember ...
+func (service SprintTaskMemberService) AddSprintActiveMember(taskList *retroSerializers.SprintTasksSerializer,
+	memberID uint,
+	retroID string,
+	sprintID string) *retroSerializers.SprintTasksSerializer {
+	db := service.DB
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]:
```suggestion
	gormDB := service.DB
```
change this.

File: apps/retrospective/services/sprint_task_member.go [hunk 8]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
...
+		Where("member_id = ?", memberID).
+		Find(&sprintMember)
+
+	sprintMemberTask.SprintMemberID = sprintMember.ID
+	for _, task := range taskList.Tasks {
+		sprintMemberTask.SprintTaskID = task.ID
+		task.SprintActiveMemberTime = 0
+		member, _, err := service.GetMember(sprintMemberTask, memberID, retroID, sprintID)
+
+		if err == nil {
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]: handle error.

File: apps/retrospective/services/sprint_task_member.go [hunk 9]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
...
+	sprintMemberTask := retroModels.SprintMemberTask{}
+	sprintMember := retroModels.SprintMember{}
+
+	db.Model(&retroModels.SprintMember{}).
+		Where("sprint_members.deleted_at IS NULL").
+		Where("sprint_id = ?", sprintID).
+		Where("member_id = ?", memberID).
+		Find(&sprintMember)
+
+	sprintMemberTask.SprintMemberID = sprintMember.ID
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]: please, fetch this from DB.

File: apps/retrospective/services/sprint_task_member.go [hunk 10]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
...
+	for _, task := range taskList.Tasks {
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]:
```suggestion
	for _, sprintTask := range taskList.Tasks {
```

File: apps/retrospective/services/sprint_task_member.go [hunk 11]
```diff
@@ -63,6 +63,34 @@ func (service SprintTaskMemberService) GetMembers(
 	return members, http.StatusOK, nil
 }
 
+// AddSprintActiveMember ...
+func (service SprintTaskMemberService) AddSprintActiveMember(taskList *retroSerializers.SprintTasksSerializer,
+	memberID uint,
+	retroID string,
+	sprintID string) *retroSerializers.SprintTasksSerializer {
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]: Please, return status and error also.

File: servers/server.go [hunk 12]
```diff
@@ -139,12 +139,12 @@ func (a *App) SetRoutes() {
 		TrailService:                 trailService}
 	sprintNoteController.Routes(sprintNoteRoute)
 
+	taskMemberService := retrospectiveServices.SprintTaskMemberService{DB: a.DB}
 	taskService := retrospectiveServices.SprintTaskService{DB: a.DB}
 	taskRoute := sprintRoute.Group(":sprintID/tasks")
-	tasksController := apiControllers.SprintTaskController{SprintTaskService: taskService, PermissionService: permissionService, TrailService: trailService}
+	tasksController := apiControllers.SprintTaskController{SprintTaskService: taskService, SprintTaskMemberService: taskMemberService, PermissionService: permissionService, TrailService: trailService}
```
Comment (chahat-arora) [inline servers/server.go]:
```suggestion
	tasksController := apiControllers.SprintTaskController{
	    SprintTaskService:       taskService,
	    SprintTaskMemberService: taskMemberService,
	    PermissionService:       permissionService,
	    TrailService:            trailService,
	}
```

File: apps/retrospective/services/sprint_task_member.go (hunk 4 above)
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task_member.go]: Please, use a appropriate function name.

File: apps/retrospective/serializers/sprint_task.go (hunk 1 above)
Comment (chahat-arora) [inline apps/retrospective/serializers/sprint_task.go]:
```suggestion
	SprintCurrentMemberTime uint   // Time spent on the task by the Current Login Member in the sprint
```
or something more relevant.

**Review by chahat-arora — CHANGES_REQUESTED** [review]

File: apps/retrospective/serializers/sprint_task.go (hunk 1 above)
Comment (Ravi2714) [reply apps/retrospective/serializers/sprint_task.go]: changed

File: apps/retrospective/services/sprint_task_member.go (hunk 2 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed this

File: apps/retrospective/services/sprint_task_member.go (hunk 3 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed

File: apps/retrospective/services/sprint_task_member.go (hunk 4 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: done

File: controllers/v1/sprint_task.go (hunk 5 above)
Comment (Ravi2714) [reply controllers/v1/sprint_task.go]: changed

File: apps/retrospective/services/sprint_task_member.go (hunk 6 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed this code

File: apps/retrospective/services/sprint_task_member.go (hunk 7 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed this code

File: apps/retrospective/services/sprint_task_member.go (hunk 8 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed this code

File: apps/retrospective/services/sprint_task_member.go (hunk 9 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed this code

File: apps/retrospective/services/sprint_task_member.go (hunk 10 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed

File: apps/retrospective/services/sprint_task_member.go (hunk 11 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed

File: servers/server.go (hunk 12 above)
Comment (Ravi2714) [reply servers/server.go]: changed

File: apps/retrospective/services/sprint_task_member.go (hunk 4 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task_member.go]: removed this function

File: apps/retrospective/serializers/sprint_task.go (hunk 1 above)
Comment (Ravi2714) [reply apps/retrospective/serializers/sprint_task.go]: changed

File: controllers/v1/sprint_task.go [hunk 13]
```diff
@@ -13,9 +13,9 @@ import (
 
 // SprintTaskController ...
 type SprintTaskController struct {
-	SprintTaskService retroServices.SprintTaskService
-	PermissionService retroServices.PermissionService
-	TrailService      retroServices.TrailService
+	SprintTaskService       retroServices.SprintTaskService
```
Comment (chahat-arora) [inline controllers/v1/sprint_task.go]: Run `make fmt`.

File: apps/retrospective/services/sprint_task.go [hunk 14]
```diff
@@ -232,10 +240,18 @@ func (service SprintTaskService) tasksForCurrentAndPrevSprint(retroID string, sp
 }
 
 // tasksWithTimeDetailsForCurrentAndPrevSprint ...
-func (service SprintTaskService) tasksWithTimeDetailsForCurrentAndPrevSprint(retroID string, sprintID string, sprintTaskID *string) *gorm.DB {
+func (service SprintTaskService) tasksWithTimeDetailsForCurrentAndPrevSprint(retroID string, sprintID string, sprintTaskID *string, memberID uint) *gorm.DB {
 
 	db := service.DB
 
+	sprintCurrentMemberTable := service.tasksForCurrentAndPrevSprint(retroID, sprintID).
```
Comment (chahat-arora) [inline apps/retrospective/services/sprint_task.go]: store this as variable and reuse it in other places.

**Review by chahat-arora — CHANGES_REQUESTED** [review]

File: controllers/v1/sprint_task.go (hunk 13 above)
Comment (Ravi2714) [reply controllers/v1/sprint_task.go]: done

File: apps/retrospective/services/sprint_task.go (hunk 14 above)
Comment (Ravi2714) [reply apps/retrospective/services/sprint_task.go]: done

**Review by chahat-arora — APPROVED** [review]

**Review by himanshu-garg-165 — APPROVED** [review]

**Pull request merged**